src/Microsoft.HttpRepl/FileSystem/RealFileSystem.cs

```diff
@@ -40,9 +42,28 @@ public void DeleteFile(string path)
             File.Delete(path);
         }
 
-        public string GetTempFileName()
+        public string GetTempFileName(string fileExtension)
         {
-            return Path.GetTempFileName();
+            if (fileExtension is null)
+            {
+                throw new ArgumentNullException(nameof(fileExtension));
+            }
+
+            if (!fileExtension.StartsWith(".", StringComparison.Ordinal))
```

Comment:
What should it do if `fileExtension == "."`?  Is that a valid input?

Reply:
As discussed, changed it to disallow that.